### PR TITLE
#6030 Add throwArgBlank for ArgumentUtils

### LINKED ISSUE #6030: Add `throwArgBlank` to `ArgumentUtils`
### Problem

`CommonUtils` provides a function `throwArgBlank` to ensure that the arg is not blank. 

### Solution

Add this functionality to `ArgumentUtils`

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by timo0**
Closes #6030

### CODE REVIEW

File: platform-sdk/swirlds-base/src/main/java/com/swirlds/base/ArgumentUtils.java [hunk 1]
```diff
@@ -40,4 +40,19 @@ public static <T> T throwArgNull(final T argument, final String argumentName) {
...
+     * Throw an {@link IllegalArgumentException} if the supplied {@code String} is blank.
+     *
+     * @param argument     the argument checked
+     * @param argumentName the name of the argument
+     */
+    public static String throwArgBlank(final String argument, final String argumentName) {
+        throwArgNull(argument, argumentName);
+        if (argument.isBlank()) {
+            throw new IllegalArgumentException(
+                    String.format("The supplied argument '%s' cannot be blank!", argumentName));
```
Comment (jsync-swirlds) [inline platform-sdk/swirlds-base/src/main/java/com/swirlds/base/ArgumentUtils.java:54]: String can be a constant if modified as follows:
```java
private static final String BLANK_ARG_FORMAT = "The supplied argument '%s' cannot be blank.";

throw new IllegalArgumentException(BLANK_ARG_FORMAT.formatted(argumentName));
```

**Review by jsync-swirlds — DISMISSED** [review]
Minor suggestion, otherwise looks good.

**Review by edward-swirldslabs — APPROVED** [review]

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/utility/CommonUtils.java [hunk 2]
```diff
@@ -255,7 +255,9 @@ public static <T> T throwArgNull(final T arg, final String argName) {
      * @param arg     the argument checked
      * @param argName the name of the argument
      * @see StringUtils#isBlank(CharSequence)
+     * @deprecated use {@link com.swirlds.base.ArgumentUtils#throwArgBlank(String, String)} instead
```
Comment (edward-swirldslabs) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/utility/CommonUtils.java:258]: @hendrikebbers, Given these conversations and this decision, should we change the comment about what to use instead, or change the implementation of `ArgumentUtils.throwArgBlank` to be an inlined call backed by `Objects.requireNonNull`?
* [conversation 1](https://swirldslabs.slack.com/archives/C04RFBCCP8B/p1681139457284359) 
* [conversation 2](https://swirldslabs.slack.com/archives/C03DU0QGE2G/p1681242564890119)
* [Decision/Outcome](https://www.notion.so/swirldslabs/007-Java-Code-Style-a0130a7059b044f68447b9ca3a5242b3?pvs=4#fadddb0f9c974f06a8fecccff951eb96)

**Review by edward-swirldslabs — DISMISSED** [review]

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/utility/CommonUtils.java (hunk 2 above)
Comment (hendrikebbers) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/utility/CommonUtils.java:258]: Yes but we should not do that in this PR. Maybe you can create a new one?

Comment (hendrikebbers) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/utility/CommonUtils.java:258]: And maybe we deprecate `ArgumentUtils.throwArgBlank`?

**Review by hendrikebbers — APPROVED** [review]

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/utility/CommonUtils.java (hunk 2 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/utility/CommonUtils.java:258]: I created an issue for it: https://github.com/hashgraph/hedera-services/issues/6139

I'm good with resolving this conversation.

File: platform-sdk/swirlds-base/src/main/java/com/swirlds/base/ArgumentUtils.java [hunk 3]
```diff
@@ -36,7 +39,21 @@ private ArgumentUtils() throws IllegalAccessException {
...
+    }
+
+    /**
+     * Throw an {@link IllegalArgumentException} if the supplied {@code String} is blank.
+     *
+     * @param argument     the argument checked
+     * @param argumentName the name of the argument
+     */
+    public static String throwArgBlank(final String argument, final String argumentName) {
+        throwArgNull(argument, argumentName);
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-base/src/main/java/com/swirlds/base/ArgumentUtils.java:54]: We decided to use Objects.requireNonNull

File: platform-sdk/swirlds-base/src/main/java/com/swirlds/base/ArgumentUtils.java [hunk 4]
```diff
@@ -36,7 +39,21 @@ private ArgumentUtils() throws IllegalAccessException {
      */
     public static <T> T throwArgNull(final T argument, final String argumentName) {
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-base/src/main/java/com/swirlds/base/ArgumentUtils.java:40]: null annotations missing

**Review by jsync-swirlds — APPROVED** [review]

**Review by edward-swirldslabs — APPROVED** [review]

**Pull request merged**